pkg/summarizer/summary_test.go

```diff
@@ -1267,35 +1277,181 @@ func TestStatusMessage(t *testing.T) {
+					Name: "four pass",
+					Results: []int32{
+						int32(statepb.Row_PASS), 4,
+					},
+				},
+				{
+					Name: "four fail",
+					Results: []int32{
+						int32(statepb.Row_FAIL), 4,
+					},
+				},
+			},
+			passingCols:       0,
+			filledCols:        4,
+			passingCells:      4,
+			filledCells:       8,
+			expectedFailRatio: []float32{.5, .5, .5, .5},
+		},
+		{
+			name:   "no result in every column",
+			cols:   3,
+			recent: 3,
+			rows: []*statepb.Row{
+				{
+					Name:    "always empty",
+					Results: []int32{int32(statepb.Row_NO_RESULT), 3},
+				},
+				{
+					Name: "first empty",
+					Results: []int32{
+						int32(statepb.Row_NO_RESULT), 1,
+						int32(statepb.Row_PASS), 2,
+					},
+				},
+				{
+					Name:    "always empty",
+					Results: []int32{int32(statepb.Row_NO_RESULT), 3},
+				},
+			},
+			passingCols:       2,
+			filledCols:        2,
+			passingCells:      2,
+			filledCells:       2,
+			expectedFailRatio: []float32{0, 0, 0},
+		},
+		{
+			name:   "only no result",
+			cols:   3,
+			recent: 3,
+			rows: []*statepb.Row{
+				{
+					Name:    "always empty",
+					Results: []int32{int32(statepb.Row_NO_RESULT), 3},
+				},
+			},
+			passingCols:       0,
+			filledCols:        0,
+			passingCells:      0,
+			filledCells:       0,
+			expectedFailRatio: []float32{0, 0, 0},
+		},
+		{
+			name:   "Pass with skips",
+			cols:   3,
+			recent: 3,
+			rows: []*statepb.Row{
+				{
+					Name:    "always empty",
+					Results: []int32{int32(statepb.Row_PASS_WITH_SKIPS), 3},
+				},
+				{
+					Name:    "all pass",
+					Results: []int32{int32(statepb.Row_PASS), 3},
 				},
 			},
-			passingCols:  2,
-			filledCols:   2,
-			passingCells: 2,
-			filledCells:  2,
+			passingCols:       3,
+			filledCols:        3,
+			passingCells:      6,
+			filledCells:       6,
+			expectedFailRatio: []float32{0, 0, 0},
+		},
+		{
+			name:   "Pass with errors",
+			cols:   3,
+			recent: 3,
+			rows: []*statepb.Row{
+				{
+					Name:    "always empty",
+					Results: []int32{int32(statepb.Row_PASS_WITH_ERRORS), 3},
+				},
+				{
+					Name:    "all pass",
+					Results: []int32{int32(statepb.Row_PASS), 3},
+				},
+			},
+			passingCols:       3,
+			filledCols:        3,
+			passingCells:      6,
+			filledCells:       6,
+			expectedFailRatio: []float32{0, 0, 0},
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			expected := makeShim(tc.passingCols, tc.filledCols, tc.passingCells, tc.filledCells, tc.expectedFailRatio)
+			actual := makeShim(gridMetrics(tc.cols, tc.rows, tc.recent))
+			assert.Equal(t, expected, actual, fmt.Sprintf("%s != expected %s", actual, expected))
+		})
+	}
+}
+
+func TestStatusMessage(t *testing.T) {
+	cases := []struct {
+		name             string
+		passingCols      int
+		completedCols    int
+		passingCells     int
+		filledCells      int
+		expectedOverride string
+	}{
+		{
+			name:             "no filledCells",
```

Comment:
This is the same as "all no result" below I think?

Reply:
true... I was trying to be explicit and not thinking about the fact that it is just the base value for int